File: android/app/src/main/kotlin/dev/tcode/thinmpf/api/HostSongApiImpl.kt
package dev.tcode.thinmpf.api

import HostSongApi
import Song
import android.content.Context
import dev.tcode.thinmpf.repository.SongRepository

class HostSongApiImpl(private val context: Context) : HostSongApi {
    override fun findAll(): List<Song> {
        val repository = SongRepository(context)
        val result = repository.findAll()

        return result.map {
            Song(it.id, it.name, it.artistName, it.albumId)
        }
    }
}

File: ios/Runner/Api/HostSongApiImpl.swift
//
//  HostSongApiImpl.swift
//  Runner
//
//  Created by tk on 2024/01/25.
//

import Foundation

class HostSongApiImpl: HostSongApi {
    func findAll() throws -> [Song] {
        let repository = SongRepository()
        let songs = repository.findAll()

        return songs.map { song in
            return Song(id: song.id, title: song.primaryText, artist: song.secondaryText)
        }
    }
}
